morphia/src/test/java/org/mongodb/morphia/IndexHelperTest.java

```diff
@@ -0,0 +1,443 @@
+        assertEquals(new BsonDocument()
+                         .append("text", new BsonString("text"))
+                         .append("nest", new BsonInt32(-1)),
+                     keys);
+    }
+
+    @Test
+    public void createIndex() {
+        MongoCollection<Document> collection = getDatabase().getCollection("indexes");
+        MappedClass mappedClass = getMorphia().getMapper().getMappedClass(IndexedClass.class);
+
+        indexHelper.createIndex(collection, mappedClass, false);
+        List<DBObject> indexInfo = getDs().getCollection(IndexedClass.class)
+                                          .getIndexInfo();
+        assertEquals("Should have 5 indexes", 5, indexInfo.size());
+        for (DBObject dbObject : indexInfo) {
+            String name = dbObject.get("name").toString();
+            if (name.equals("latitude_1")) {
+                assertEquals(parse("{ 'latitude' : 1 }"), dbObject.get("key"));
+            } else if (name.equals("behind_interface")) {
+                assertEquals(parse("{ 'nest.name' : -1} "), dbObject.get("key"));
+                assertEquals(parse("{ 'locale' : 'en' , 'caseLevel' : false , 'caseFirst' : 'off' , 'strength' : 2 , 'numericOrdering' :"
+                                       + " false , 'alternate' : 'non-ignorable' , 'maxVariable' : 'punct' , 'normalization' : false , "
+                                       + "'backwards' : false , 'version' : '57.1'}"), dbObject.get("collation"));
+            } else if (name.equals("nest.name_1")) {
+                assertEquals(parse("{ 'nest.name' : 1} "), dbObject.get("key"));
+            } else if (name.equals("searchme")) {
+                assertEquals(parse("{ 'text' : 10 }"), dbObject.get("weights"));
+            } else {
+                if (!"_id_".equals(dbObject.get("name"))) {
+                    throw new MappingException("Found an index I wasn't expecting:  " + dbObject);
+                }
+            }
+
+        }
+
+    }
+
+    @Test
+    public void findField() {
+        MappedClass mappedClass = getMorphia().getMapper().getMappedClass(IndexedClass.class);
+
+        assertEquals("name", indexHelper.findField(mappedClass, new IndexOptionsBuilder(), singletonList("indexName")));
+        assertEquals("name", indexHelper.findField(mappedClass, new IndexOptionsBuilder(), singletonList("name")));
+        assertEquals("nest.name", indexHelper.findField(mappedClass, new IndexOptionsBuilder(), asList("nested", "name")));
+        assertEquals("nest.name", indexHelper.findField(mappedClass, new IndexOptionsBuilder(), asList("nest", "name")));
+
+        try {
+            assertEquals("nest.whatsit", indexHelper.findField(mappedClass, new IndexOptionsBuilder(), asList("nest", "whatsit")));
+            fail("Should have failed on the bad index path");
+        } catch (MappingException e) {
+            // alles ist gut
+        }
+        assertEquals("nest.whatsit.nested.more.deeply.than.the.object.model",
```

Comment:
Is it the disable validation flag that allows this one to pass rather than the previous example?